### DISCUSSION TITLE
Add AOP to the language, please

### DISCUSSION THREAD

**Discussion by pelletburner** [dotnet/csharplang#6703] [Language Ideas]
We should get language integrated features for AOP, based on attributes.

Reasoning:

Things like "INotifyPropertyChanged" are downright horrible to implement, getting management sign off for paid packages like PostSharp is not always easy and the language should support a feature like this anyway. Also it's easier to get people on board a standard language feature that is simply expected to be used instead of "just another nuget package".

Would like to add that even scripting languages like Python and Ruby support this out of the box.

Also C# is already a language that does a lot to reduce the amount of boilerplate code to be written (when compared to, for example, Java). Lets take it a BIG step further.

Implementation idea:

Don't do weaving, the feature should be part of the compiler itself and work with a mechanism similar, but not exactly the same, as attributes. Same syntax would be great, but there should be different base classes ("MethodInvocationAttribute" or so).

**Comment by HaloFour** [comment]
See: #107

As of now the team does not appear to be interested in supporting something like this.

**Comment by 333fred** [comment]
For INPC specifically, my suggestion (which I use myself) is to use CommunityToolkit.Mvvm,which includes a source generator to generate INPC implementations for you.
